### DISCUSSION TITLE
Keycloak database inside of Hasura

### DISCUSSION THREAD

**Discussion by Keyruu** [hasura/graphql-engine#8684] [Question] [answered]
Hi guys! I was just looking into using Keycloak with Hasura: https://hasura.io/learn/graphql/hasura-authentication/integrations/keycloak/
And I noticed the mention of having to sync user information between Hasura and Keycloak...
But couldn't you just hook the Keycloak DB up to Hasura as a remote DB or even use the default database of Hasura for the Keycloak DB?
SuperTokens does the same thing if I read that correctly...
Would love to hear your thoughts on this!

**Comment by adas98012** [comment, marked as answer]
PostgreSQL and MS SQL Server are both supported databases for Keycloak, just as they both are supported databases for Hasura.  Therefore, the Keycloak database could be added as another database to Hasura, like any other database can be.  
      Alternatively, a non-remote database configured in Hasura could do double-duty as the database for Keycloak. 
      Please try out the options and let us know. Looking forward to hear about a successful outcome.
